### PR TITLE
Chalhouniversity facetoface confirmation message

### PR DESCRIPTION
**Pull request by marcghaly**
Hi @mattporritt, @danmarsden, 

this PR is a combination of https://github.com/catalyst/moodle-mod_facetoface/pull/56, https://github.com/catalyst/moodle-mod_facetoface/pull/72 and https://github.com/catalyst/moodle-mod_facetoface/pull/75. It is rebased against [MOODLE_311_STABLE](https://github.com/catalyst/moodle-mod_facetoface/tree/MOODLE_311_STABLE).

Comments of @danmarsden were addressed.

Cheers,

### CODE REVIEW

File: db/upgrade.php [hunk 1]
```diff
@@ -45,18 +45,14 @@ function facetoface_send_admin_upgrade_msg($data) {
...
     $table->data = $data;
     $table->align = array ('center', 'center', 'center');
 
-    $title    = "$SITE->fullname: Face to Face upgrade info";
-    $note = 'During the last site upgrade the face-to-face module has been modified. It now
-requires session custom fields to have unique shortnames. Since some of your
-custom fields had duplicate shortnames, they have been renamed to remove
-duplicates (see table below). This could impact on your email messages if you
-reference those custom fields in the message templates.';
+    $title    = get_string('upgradeinfofacetoface', 'mod_facetoface', $SITE->fullname);
```
Comment (danmarsden) [inline db/upgrade.php]: would be good to tidy up the spacing here $title should have one space then the =

File: lib.php [hunk 2]
```diff
@@ -2305,8 +2310,11 @@ function facetoface_send_confirmation_notice($facetoface, $session, $userid, $no
     // Set invite bit.
     $notificationtype |= MDL_F2F_INVITE;
 
+    // Set HTML Body.
+    $htmlmessage = $facetoface->confirmationmessage;
+
     return facetoface_send_notice($postsubject, $posttext, $posttextmgrheading,
-                                  $notificationtype, $facetoface, $session, $userid);
+                                  $notificationtype, $facetoface, $session, $userid,$htmlmessage);
```
Comment (danmarsden) [inline lib.php]: missing space after the comma here still.

File: mod_form.php [hunk 3]
```diff
@@ -122,9 +122,13 @@ public function definition() {
...
 
-        $mform->addElement('textarea', 'confirmationmessage', get_string('email:message', 'facetoface'), 'wrap="virtual" rows="15" cols="70"');
-        $mform->setDefault('confirmationmessage', get_string('setting:defaultconfirmationmessagedefault', 'facetoface'));
-
+        $editoroptions = ['trusttext' => true];
+        $mform->addElement('editor', 'confirmationmessage', get_string('email:message', 'facetoface'), null, $editoroptions);
+        $mform->setType('confirmationmessage', PARAM_RAW);
+        $confirmationmessagedata = [
+            'text' => get_string('setting:defaultconfirmationmessagedefault2', 'facetoface'),
+            'format' => 1
```
Comment (danmarsden) [inline mod_form.php]: Magic number? "1" - shouldn't this be FORMAT_HTML ?

File: db/upgrade.php (hunk 1 above)
Comment (marcghaly) [reply db/upgrade.php]: I am using codechecker plugin - is it supposed to find those cs errors ?

File: lib.php (hunk 2 above)
Comment (marcghaly) [reply lib.php]: fixed

File: mod_form.php (hunk 3 above)
Comment (marcghaly) [reply mod_form.php]: Fixed

File: db/upgrade.php (hunk 1 above)
Comment (marcghaly) [reply db/upgrade.php]: For some reason it wasn't caught when I ran code checker - fixed it.

**Pull request merged**